python/dgl/graphbolt/minibatch_sampler.py

```diff
@@ -0,0 +1,104 @@
+"""Minibatch Sampler"""
+
+from typing import Mapping, Optional
+
+from torch.utils.data import default_collate
+from torchdata.datapipes.iter import IterableWrapper, IterDataPipe
+from .itemset import *
+from ..batch import batch as dgl_batch
+from ..heterograph import DGLGraph
+
+__all__ = ["MinibatchSampler"]
+
+
+def _colloate(batch):
+    """Collate batch."""
+    data = next(iter(batch))
+    if isinstance(data, DGLGraph):
+        return dgl_batch(batch)
+    elif isinstance(data, Mapping):
+        raise NotImplementedError
+    return default_collate(batch)
+
+
+class MinibatchSampler(IterDataPipe):
+    """Minibatch Sampler.
+
+    Creates mini-batches of data which could be node/edge IDs, node pairs with
+    or without labels, head/tail/negative_tails, DGLGraphs and heterogeneous
+    counterparts.
+
+    Note: This class `MinibatchSampler` is not decorated with
+    `torchdata.datapipes.functional_datapipe` on purpose. This indicates it
+    does not support function-like call. But any iterable datapipes from
+    `torchdata` can be further appended.
+
+    Parameters
+    ----------
+    item_set : ItemSet or DictItemSet
```

Comment:
add an example for DictItemSet?

Reply:
`DictItemSet` is not supported yet as I'd like to make this PR small enough. I'd like to add support for `DictItemSet` in another PR.

Reply:
Remove comment related to DictItemSet in this PR and add in the other one to avoid confusion.